docs/confluence/advanced/grounding.md

```diff
@@ -0,0 +1,154 @@
+# Grounding 
+
+## Flag
+
+Grounding is on by default. To turn it off, use `--settings -smt_groundQuantifiers=false`
+
+ 
+## Grounding General Idea
+
+Grounding should make using quantifiers much easier on the solvers. For example,
+
+```
+assume forall x. f(x) = 0
+```
+
+A very natural init state axiom. Instead of giving it as is to the SMT-solvers, we automatically collect all instances of `f` in the formula we want to verify, e.g., `f(2), f(9), f(y + 3), f(z)`, and replace the original assert with:
+
+```
+assume f(2) = 0
+assume f(9) = 0
+assume f(y + 3) = 0
+assume f(z) = 0
+```
+
+Simple enough! You can of course write more complex quantified statements, and they will be grounded in a similar fashion. However, this approach has its limitations. It will never say a rule is valid when it is not (i.e., grounding is sound), but it may generate false counter-examples. Read below on how to minimize the chance of this happening. 
```

Comment:
Simple enough! You can also write more complex quantified statements, which will be grounded in a similar fashion. However, this approach has its limitations. It will never validate a rule when it is not valid (i.e., grounding is sound), but it may generate false counter-examples. Read below on how to minimize the chance of this happening.